### ISSUE TITLE
error says to set source.dir but source.dir is set

### ISSUE THREAD

**Issue by vitiral** [luarocks/luarocks#1675]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions in the [Gitter
  chat-room](https://gitter.im/luarocks/luarocks).
- Please do NOT submit duplicate bug reports. Look through the [issue
  tracker](https://github.com/luarocks/luarocks/issues) before submitting.

### Please fill out the following:
- **Platform:**
  - Linux

- **LuaRocks version:**
  - `luarocks --version`
/usr/local/bin/luarocks 3.11.0
LuaRocks main command-line interface

- **Configuration file:** default

- **LuaRocks output from when the issue occurred:**
  - Use the `--verbose` flag.
attached
[luarocks.log](https://github.com/luarocks/luarocks/files/15327161/luarocks.log)

- **Description:**
  - The steps you took in plain English to reproduce the problem.

```
$ git clone git@github.com:civboot/civlua.git
$ git checkout pkg-0.1-9
$ luarocks build lib/pkg/pkg-0.1-9.rockspec --local --verbose
Error: Could not determine source directory, please set source.dir in rockspec.=
```

The `source.dir` IS set, correctly (I think)

**Comment by vitiral** [comment]
Here's `lib/pkg/pkg-0.1-9.rockspec`

```
build = {
  modules={pkg="pkglib.lua"},
  type="builtin"
}
dependencies = {"lua ~> 5.3"}
description = {
  homepage="https://github.com/civboot/civlua/blob/main/lib/pkg/README.md",
  license="UNLICENSE",
  summary="better lua package creation and importing"
}
package = "pkg"
rockspec_format = "3.0"
source = {
  dir='lib/pkg',
  tag="pkg-0.1-9",
  url="git+http://github.com/civboot/civlua"
}
version = "0.1-9"
```

**Comment by hishamhm** [comment]
@vitiral source.dir includes the "basedir" of the repo or uncompressed package. So, please use `dir = 'civlua/lib/pkg',` and that'll fix it.

**Comment by vitiral** [comment]
Thanks! So this is intended behavior? It feels very strange to need to include the source directory, since presumably luarocks clones it from git and specifies the name right?
